lib/ShareByCircleProvider.php

```diff
@@ -98,7 +97,7 @@ public function __construct(
 		IUserManager $userManager,
 		IRootFolder $rootFolder,
 		IL10N $l10n,
-		ILogger $logger,
+		mixed $logger, // unused, only kept for compatibility with server
```

Comment:
Damn this is ugly.
Server should use DI and the app should register the service if needed.

Reply:
Yes I know but there are also other parameters where this is already done :pensive:

Reply:
But server already checks `class_exists`, meaning it can be auto-loaded, so we should theoretically be able to simply DI in server.